stl/inc/functional

```diff
@@ -1768,31 +1768,62 @@ private:
 
 // FUNCTION TEMPLATE _Build_boyer_moore_delta_2_table
 template <class _RanItPat, class _Pred_eq>
-void _Build_boyer_moore_delta_2_table(_Iter_diff_t<_RanItPat>* _Shifts, const _RanItPat _Pat_first,
+void _Build_boyer_moore_delta_2_table(_Iter_diff_t<_RanItPat>* const _Shifts, const _RanItPat _Pat_first,
     const _Iter_diff_t<_RanItPat> _Pat_size, _Pred_eq& _Eq) {
-    // builds Boyer-Moore's delta_2 table from a pattern [_Pat_first, _Pat_first + _Pat_size)
-    // pre: _Shifts is a pointer to _Pat_size _Iter_diff_t<_RanIt>s
+    // Builds Boyer-Moore's delta_2 table from a pattern [_Pat_first, _Pat_first + _Pat_size).

```

Comment:
If you're gonna change this line, fix the scope operator mismatch, right?

Reply:
`[a, b)` means inclusive, exclusive. Unless you're saying this should be inclusive for both?

Reply:
Like almost all ranges in the STL, this is indeed an inclusive-exclusive (aka half-open) range, so the `[first, last)` notation is correct and conventional as @SuperWig noted. Thank you, @mattspring, for bringing the potential typo to our attention - it is so easy to swap `]` and `)` and the difference is significant.

Reply:
Whoops, protip: don't drink and review strangers repos

Reply:
> Whoops, protip: don't drink and review strangers repos

Does this mean it's ok to drink while reviewing repos you're a member of? Asking for a friend.